plugins/modules/win_disk_facts.ps1

```diff
@@ -4,16 +4,29 @@
 # GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
 
 #Requires -Module Ansible.ModuleUtils.Legacy
+#AnsibleRequires -CSharpUtil Ansible.Basic
```

Comment:
If you are converting this to the newer style the above line `#Requires -Module Ansible.ModuleUtils.Legacy` is no longer needed but it requires a few more changes to migrate from the legacy stuff. The results are meant to be stored in `$module.Result`, `Exit-Json` and `Fail-Json` have changed to `$module.ExitJson()` and `$module.FailJson("msg")`. Unless you are willing to do all this work I would stick with the legacy style for now or address the migration to `Ansible.Basic` in a separate PR.